### ISSUE TITLE
[Bug]: Cardinality Violation in Postgres

### ISSUE THREAD

**Issue by morrislaptop** [hirethunk/verbs#17]
### What happened?

I get the following errors when trying to use Postgres. It seems to work in MySQL.

```
SQLSTATE[21000]: Cardinality violation: 7 ERROR:  ON CONFLICT DO UPDATE command cannot affect row a second time
HINT:  Ensure that no rows proposed for insertion within the same command have duplicate constrained values. (Connection: pgsql, SQL: insert into "verb_snapshots" ("created_at", "data", "id", "last_event_id", "type", "updated_at") values (2023-11-21 04:57:14, {"balance_in_cents":100000,"id":117488472096960512,"last_event_id":117488472088571904}, 117488472096960512, 117488472088571904, App\States\AccountState, 2023-11-21 04:57:14), (2023-11-21 04:57:14, {"balance_in_cents":100000,"id":117488472096960512,"last_event_id":117488472088571904}, 117488472096960512, 117488472088571904, App\States\AccountState, 2023-11-21 04:57:14) on conflict ("id") do update set "data" = "excluded"."data", "last_event_id" = "excluded"."last_event_id", "updated_at" = "excluded"."updated_at")
```

### How to reproduce the bug

Follow one of the examples and try to run the tests. 

### Package Version

0.0.5

### PHP Version

8.2.0

### Laravel Version

10.10

### Which operating systems does with happen with?

macOS

### Notes

It doesn't seem to be calling upsert twice so I'm not sure why we're getting this error.

**Comment by DanielCoulbourne** [comment]
@morrislaptop thanks for the report. Haven't played with Verbs much in PGSQL, so I'll look around.

Which example were you running into this with?

**Comment by morrislaptop** [comment]
Bank and Monopoly. I didn’t try the others.

**Comment by inxilpro** [comment]
This should be resolved by #28. Thanks again, @morrislaptop!
